test/data/resource/genesyscloud_journey_action_map/action_media_types/06_update_architect_flow_with_mapping.tf

```diff
@@ -0,0 +1,40 @@
+resource "genesyscloud_journey_action_map" "terraform_test_-TEST-CASE-" {
+  display_name          = "terraform_test_-TEST-CASE-_updated"
+  trigger_with_segments = [genesyscloud_journey_segment.terraform_test_-TEST-CASE-_action_map_dependency.id]
+  activation {
+    type = "immediate"
+  }
+  action {
+    media_type = "architectFlow"
+    architect_flow_fields {
```

Comment:
Hey guys,

Whats the status of this PR?  The limitations of the OmitEmpty are a challenge with the Go language.  Just curious as to why you feel the GO Genesys Cloud SDK is mandatory.  If the SDK is giving you problems, you can still call the endpoints directly.  We have done this a few times with beta endpoints and if SDK is blocking you can we move to an endpoint to get his PR moving.

Thanks,
     John

Reply:
Hi John,

Our goal is to provide a "fully" functional terraform interface, where create/update and delete also supported. Since the Go SDK does not provide proper option for update or delete fields properly our guess was that Go SDK should be fix.

Probably we could call directly API endpoint but Go SDK will still be broken. What is the advantage to have an SDK which does not support update / delete properties?
I think Go SDK should be fixed the allow sending null for fields where the API endpoints allow it for update/delete purpose.

As far as I know the Go SDK generated from the Public API swagger schema. There is a property called `required`, this could indicate which fields could skip `omitEmpty`. Of course it could be more complicated, not familiar with all details.

My main concern with current Go SDK that I'm not even able to remove a `Description` property from any resource because `omitEmpty` not allowing to send me a null value.

In terraform there is a workaround for this cases, but this will mean deleting `Description` property requires to recreate the whole resource. If the resource has dependencies it will get complicated and did not mentioned that this way we lose history too.

The `Description` example the smallest issue, just an easy use case to represent the problem.

Regards,
Istvan

Reply:
Hi Istvan,

I can appreciate your frustration, but it has nothing to do with the Go SDK.  The core Golang JSON libraries automatically remove null values (e.g. omitEmpty).   We use the Swagger docs to generate structs and the omitEmpty tags are there to make sure we don't send a field if the value is empty string.  Otherwise, APIs can see the empty string and try to update it even if the user did not intend to have the field populated.

Most APIs end up implementing something like this: https://themirgleich.medium.com/how-golangs-omitempty-can-confuse-the-google-apis-c0e46d22d9ff.  We might be able to do something using custom swagger fields and generating something like that for the methods that are annotated but that is not trivial work and we do not have the resources, bandwidth, or priority to try and do something like that for this particular use case.  

I really think your best option is to build and manipulate the JSON manually to set any fields you want null.

Regards,
    John

Reply:
Hi John,

Removing `omitEmpty` from proper places in the Go SDK, solve the problem. I tested and validated. Ronan was also working on a general solution to remove `omitEmpty` property where not appropriate.

In my case:
![image](https://user-images.githubusercontent.com/107180950/194324774-93dfe9f6-e995-4f04-91b9-dcd38890563a.png)
where for example I removed `omitEmpty`:
![image](https://user-images.githubusercontent.com/107180950/194325014-42f7a263-4b1f-4744-a384-4a550ea25ce7.png)

These `omitEmpty` configs are added at code generation from Swagger schema. This generation should be updated to have proper field annotation. Ronan managed to have this in one his branches.

Regards,
Istvan

Reply:
Ronan's branch, used for develop: https://github.com/ronanwatkins/platform-client-sdk-go